### PR TITLE
Fixes and updating version number

### PR DESCRIPTION
**Pull request by kratman**
Changes:
- Update version number to v0.5.0 so that users can pin to `bpx<0.5.0` if required.
- Fixed some typing errors due to paths/strings
- Switched to the default JSON implementation since there did not seem to be a need to install the other dependency
- Switched `copy.copy()` to `copy.deepcopy()` as an extra layer of safety since all of the tests interact with the same dictionary
- Disabled fail-fast since the tests are quick

### CODE REVIEW

File: bpx/function.py [hunk 1]
```diff
@@ -87,26 +86,10 @@ def to_python_function(self, preamble: str | None = None) -> Callable:
...
-            tmp.close()
-            Path(tmp.name).unlink(missing_ok=True)
-            if module.__cached__:
-                cached_file = Path(module.__cached__)
-                cached_path = cached_file.parent
-                cached_file.unlink(missing_ok=True)
-                if not any(cached_path.iterdir()):
-                    cached_path.rmdir()
-
-        # return the new function object
```
Comment (kratman) [inline bpx/function.py:111]: This was causing failures due to cached files not being found. It did not seem necessary

File: bpx/parsers.py [hunk 2]
```diff
@@ -26,13 +31,13 @@ def parse_bpx_obj(bpx: dict, v_tol: float = 0.001) -> BPX:
     return BPX.model_validate(bpx)
 
 
-def parse_bpx_file(filename: str, v_tol: float = 0.001) -> BPX:
+def parse_bpx_file(filename: str | Path, v_tol: float = 0.001) -> BPX:
```
Comment (kratman) [inline bpx/parsers.py:34]: This makes the file name more flexible

File: bpx/parsers.py [hunk 3]
```diff
@@ -42,18 +47,12 @@ def parse_bpx_file(filename: str, v_tol: float = 0.001) -> BPX:
     BPX: :class:`bpx.BPX`
         a parsed BPX model
     """
-
-    from pathlib import Path
-
-    bpx = ""
-    if filename.endswith((".yml", ".yaml")):
+    if str(filename).endswith((".yml", ".yaml")):
```
Comment (kratman) [inline bpx/parsers.py:50]: Related to the path check

**Review by kratman — COMMENTED** [review]
Explanations

**Review by rtimms — APPROVED** [review]

**Pull request merged**